private/buf/bufstudioagent/bufstudioagent_test.go

```diff
@@ -193,14 +194,31 @@ func testPlainPostHandlerErrors(t *testing.T, upstreamServer *httptest.Server) {
 	})
 
 	t.Run("invalid_upstream", func(t *testing.T) {
-		listener, err := net.Listen("tcp", "127.0.0.1:")
+		tcpAddr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:")
 		require.NoError(t, err)
+		listener, err := net.ListenTCP("tcp", tcpAddr)
+		require.NoError(t, err)
+		done := make(chan struct{})
+		t.Cleanup(func() {
+			done <- struct{}{}
+		})
 		go func() {
-			conn, err := listener.Accept()
-			require.NoError(t, err)
-			require.NoError(t, conn.Close())
+			for {
+				select {
+				case <-done:
+					listener.Close()
+					return
+				default:
+					err := listener.SetDeadline(time.Now().Add(time.Second))
+					require.NoError(t, err)
+					conn, err := listener.Accept()
+					if err != nil {
+						continue
+					}
+					require.NoError(t, conn.Close())
```

Comment:
`require` is not safe from go routines: https://pkg.go.dev/testing#T.FailNow

Reply:
Good catch!

Reply:
Updated them to `continue`

Reply:
You can use `t.Error` safely as that will call fail not failnow.

Reply:
Fixed in [4852e27](https://github.com/bufbuild/buf/pull/3113/commits/4852e27e51e59999c54730a4eb23f841423474c4)